rpc/src/rewards_recorder_service.rs

```diff
@@ -38,11 +42,25 @@ impl RewardsRecorderService {
                     if exit.load(Ordering::Relaxed) {
                         break;
                     }
-                    if let Err(RecvTimeoutError::Disconnected) = Self::write_rewards(
-                        &rewards_receiver,
-                        &max_complete_rewards_slot,
-                        &blockstore,
-                    ) {
+
+                    let rewards = match rewards_receiver.try_recv() {
+                        Ok(rewards) => rewards,
+                        Err(TryRecvError::Empty) => {
+                            sleep(TRY_RECV_INTERVAL);
+                            continue;
+                        }
+                        Err(err @ TryRecvError::Disconnected) => {
+                            info!("RewardsRecorderService is stopping because: {err}");
```

Comment:
warn or error level?

Reply:
The service on the other side of the channel (`ReplayStage`) could stop in a "natural" way (ie exit flag has been set), so I kept it at info. If `ReplayStage` stops for a bad reason, it should be on the hook for an `error` log